### PR TITLE
scheduler: use async snapshot

### PR DESCRIPTION
**Pull request by overvenus**
## What have you changed? (mandatory)

Scheduler takes snapshot via the `async_snapshot` API. 

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit tests.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Refer to a related PR or issue link (optional)

#2111

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -1214,82 +1066,51 @@ impl<E: Engine> Scheduler<E> {
...
-                to_be_write,
-                rows,
-            } => self.on_write_prepare_finished(cid, ctx, tag, pr, to_be_write, rows),
-            Msg::WritePrepareFailed { cid, err } => self.on_write_prepare_failed(cid, err),
-            Msg::WriteFinished {
-                cid, pr, result, ..
-            } => self.on_write_finished(cid, pr, result),
-        }
+    fn run(&mut self, _: Msg<E>) {
+        unimplemented!()
```
Comment (Hoverbear) [inline src/storage/txn/scheduler.rs]: Is this still a WIP?

Comment (overvenus) [reply src/storage/txn/scheduler.rs]: Oh, I should change it to `panic`.

File: src/storage/txn/scheduler.rs [hunk 2]
```diff
@@ -1214,82 +1066,51 @@ impl<E: Engine> Scheduler<E> {
...
-                to_be_write,
-                rows,
-            } => self.on_write_prepare_finished(cid, ctx, tag, pr, to_be_write, rows),
-            Msg::WritePrepareFailed { cid, err } => self.on_write_prepare_failed(cid, err),
-            Msg::WriteFinished {
-                cid, pr, result, ..
-            } => self.on_write_finished(cid, pr, result),
-        }
+    fn run(&mut self, _: Msg<E>) {
+        panic!("Shouldn't call Scheduler::run directly");
```
Comment (Connor1996) [inline src/storage/txn/scheduler.rs]: I still prefer implementing `run()` and call it in `run_batch()` which is consistent with other codes

File: src/storage/txn/scheduler.rs [hunk 3]
```diff
@@ -910,7 +859,7 @@ impl<E: Engine> Scheduler<E> {
     /// This method will try to acquire all the necessary latches. If all the necessary latches are
     /// acquired, the method initiates a get snapshot operation for furthur processing; otherwise,
     /// the method adds the command to the waiting queue(s). The command will be handled later in
-    /// `lock_and_register_get_snapshot` when its turn comes.
+    /// `lock_and_get_snapshot` when its turn comes.
```
Comment (Connor1996) [inline src/storage/txn/scheduler.rs]: Seems that "The command will be handled later in..." is stale, the command is actually handled in on_snapshot_finished

File: src/storage/txn/scheduler.rs [hunk 4]
```diff
@@ -910,7 +859,7 @@ impl<E: Engine> Scheduler<E> {
     /// This method will try to acquire all the necessary latches. If all the necessary latches are
     /// acquired, the method initiates a get snapshot operation for furthur processing; otherwise,
```
Comment (Connor1996) [inline src/storage/txn/scheduler.rs]: furthur -> further

**Pull request merged**